engine/src/EntityManager.cpp

```diff
@@ -0,0 +1,72 @@
+//
+// Created by tdancois on 8/4/16.
+//
+
+#include "EntityManager.h"
+#include "TextureManager.h"
+#include "AresException.h"
+
+EntityManager *EntityManager::instance;
+
+EntityManager::EntityManager() : isPlayerSet(false) {}
+
+EntityManager *EntityManager::getInstance() {
+    if (!instance) instance = new EntityManager();
+    return instance;
+}
+
+void EntityManager::addNewPlayer(const unsigned int &id, const std::string &character_texture) {
+    if (isPlayerSet) throw new EntityManagerException("addNewPlayer", "A player is already set");
+    if (entities.count(id) != 0) throw new EntityManagerException("addNewPlayer", "A player is already set");
```

Comment:
It is normal to have the same message twice ?

And please, stop doing 
```c++
if (condition) statement
```
and do instead
```c++
if(condition) {
    statement
}
```
It deceived me several times :)